src/common/attachments/uploadAttachmentsDialog.component.tsx

```diff
@@ -67,39 +74,61 @@ const UploadAttachmentsDialog = (props: UploadAttachmentsDialogProps) => {
       .use(ProgressBar)
   );
 
+  // This is necessary to prevent multiple calls of the delete endpoint.
+  const deletedFileIds = React.useRef(new Set<string>());
+
   const updateFileMetadata = React.useCallback(
-    (
+    async (
       file?: UppyFile<UppyUploadMetadata, AwsBody>,
-      deleteMetadata?: boolean
+      deleteMetadata: boolean = false
     ) => {
-      const id = fileMetadataMap[file?.id ?? ''];
+      const fileId = file?.id;
+      if (!fileId) return;
+
+      const id = fileMetadataMap[fileId];
+
       if (id) {
-        if (deleteMetadata) {
-          // TODO: Implement logic to delete metadata using id
-          // If metadata exists for the given id, remove it from the api
-          // If not, do nothing and exit the function
+        if (deleteMetadata && !deletedFileIds.current.has(fileId)) {
+          deletedFileIds.current.add(fileId);
+          try {
+            await deleteAttachment(id);
+          } catch (error: any) {
+            handleIMS_APIError(error);
+          };
```

Comment:
```suggestion
          await deleteAttachment(id).catch((error: AxiosError) => {
            handleIMS_APIError(error);
          });
```

Reply:
i never wanted to put the type as any but it was complaining if i put AxiosError or no type at all. Not sure how i forgot to use .catch instead of try catch